### PR TITLE
MDEV-13010 NEXTVAL should increment only once for a given row

### PR DESCRIPTION
**Pull request by robertbindar**

### CODE REVIEW

File: sql/sql_lex.cc [hunk 1]
```diff
@@ -756,6 +756,9 @@ void LEX::start(THD *thd_arg)
   vers_conditions.empty();
 
   is_lex_started= TRUE;
+
+  seq_already_incr= false;
+
```
Comment (montywi) [inline sql/sql_lex.cc]: Remove extra new lines here. There is no reason to single out this particular assignment.

File: sql/sql_lex.cc [hunk 2]
```diff
@@ -7853,6 +7853,23 @@ my_var *LEX::create_outvar(THD *thd,
...
+  List_iterator<Item> li(current_select->item_list);
+  Item *item;
+
+  while ((item= li++)) {
+    if (item->type() == Item::FUNC_ITEM &&
+        static_cast<Item_func*>(item)->functype() == Item_func::NEXTVAL_FUNC &&
+        !my_strcasecmp(table_alias_charset,
+                       static_cast<Item_func_nextval*>(item)->table_name(),
+                       table_name.str))
+    { return true; }
```
Comment (montywi) [inline sql/sql_lex.cc]: Remove {} around return true

File: sql/sql_lex.cc [hunk 3]
```diff
@@ -7853,6 +7853,23 @@ my_var *LEX::create_outvar(THD *thd,
 }
 
 
+bool LEX::sequence_nextval_item_found(const LEX_CSTRING &table_name)
```
Comment (montywi) [inline sql/sql_lex.cc]: Add a function comment

File: sql/sql_lex.cc [hunk 4]
```diff
@@ -7853,6 +7853,23 @@ my_var *LEX::create_outvar(THD *thd,
...
+{
+  List_iterator<Item> li(current_select->item_list);
+  Item *item;
+
+  while ((item= li++)) {
+    if (item->type() == Item::FUNC_ITEM &&
+        static_cast<Item_func*>(item)->functype() == Item_func::NEXTVAL_FUNC &&
+        !my_strcasecmp(table_alias_charset,
```
Comment (montywi) [inline sql/sql_lex.cc]: You have to check both database and table name

**Comment by LinuxJedi** [comment]
Based on Monty's last comment, I'm going to close this one for now. For starters, it will need refactoring for 11.2, but it will also need to be implemented differently to solve all the problems.

If it is possible to base a final solution based on this, it can be reopened at any time.

**Pull request closed without merging**